src/middlewared/middlewared/plugins/system.py

```diff
@@ -1433,6 +1433,12 @@ async def _event_system(middleware, event_type, args):
     global SYSTEM_SHUTTING_DOWN
     if args['id'] == 'ready':
         SYSTEM_READY = True
+        if os.path.exists(FIRST_INSTALL_SENTINEL):
```

Comment:
Cant BSD reach this code?

Reply:
It most certainly will, thank you for pointing it out :)